omod-1.12/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_12/OrderGroupController1_12Test.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_12;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.Order;
+import org.openmrs.OrderGroup;
+import org.openmrs.OrderSet;
+import org.openmrs.Patient;
+import org.openmrs.api.OrderService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_12;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+public class OrderGroupController1_12Test extends MainResourceControllerTest {
+	
+	private OrderService orderService;
+	
+	@Before
+	public void init() throws Exception {
+		orderService = Context.getOrderService();
+		executeDataSet(RestTestConstants1_12.ORDER_GROUP_TEST_DATA_SET);
+	}
+	
+	@Override
+	public String getURI() {
+		return "ordergroup";
+	}
+	
+	@Override
+	public String getUuid() {
+		return RestTestConstants1_12.ORDER_GROUP_UUID;
+	}
+	
+	@Override
+	public long getAllCount() {
+		return 0;
+	}
+	
+	@Test
+	public void shouldCreateNewOrderGroup() throws Exception {
```

Comment:
Shouldn't you assert that the count before creation of the new order group is less than that after?

Reply:
here , am only testing that a new order group was created. so the order_group in this case doesnt exist innitialy. 
Unfortunately a service method gets a list of all order groups by patient or encounter doesnt exist yet.  see this  https://issues.openmrs.org/browse/TRUNK-5225
 So theres no way i can get the number of order groups before and after

Reply:
so i can only compare the number of orders before and after , for an existing OrderGroup.
like in this test  > >` shouldAddOrdersToAnExistingOrderGroup() `

But i cant compare the number or orderGroups before and after , that method doesnt exist yet